packages/app/src/lib/import.ts

```diff
@@ -425,11 +431,110 @@ export function is1Pux(file: File): boolean {
     return file.name.endsWith(".1pux");
 }
 
+async function parseBitwardenItem(
+    vaultName: string,
+    item: BitwardenItem,
+    folders: BitwardenExport["folders"]
+): Promise<VaultItem | undefined> {
+    if (!item) {
+        return;
+    }
+
+    const { parseToRowData } = await import("./bitwarden-parser");
+
+    const rowData = parseToRowData(item, [vaultName], folders);
+
+    if (!rowData) {
+        return;
+    }
+
+    const itemName = rowData.name;
+    const tags = rowData.tags.split(",");
+
+    const fields: Field[] = [];
+
+    if (rowData.username) {
+        fields.push(new Field({ name: $l("Username"), value: rowData.username, type: FieldType.Username }));
+    }
+
+    if (rowData.password) {
+        fields.push(new Field({ name: $l("Password"), value: rowData.password, type: FieldType.Password }));
+    }
+
+    if (rowData.url) {
+        fields.push(new Field({ name: $l("URL"), value: rowData.url, type: FieldType.Url }));
+    }
+
+    if (rowData.notes) {
+        fields.push(new Field({ name: $l("Notes"), value: rowData.notes, type: FieldType.Note }));
+    }
+
+    for (const extraField of rowData.extraFields) {
+        if (extraField.type === "totp") {
+            // Extract just the secret
+            try {
+                const secret = new URL(extraField.value).searchParams.get("secret");
+                if (secret) {
+                    fields.push(new Field({ name: extraField.name, value: secret, type: FieldType.Totp }));
+                }
+            } catch (error) {
+                // Do nothing
+            }
+        } else {
+            fields.push(
+                new Field({ name: extraField.name, value: extraField.value, type: extraField.type as FieldType })
+            );
+        }
+    }
+
+    return createVaultItem({ name: itemName, fields, tags });
+}
```

Comment:
Why is this not in `lib/bitwarden-parser`?

Reply:
It can be moved later, no real reason, it was just consistency with the 1pux importer, but that can be moved too.